misk-jooq/src/main/kotlin/misk/jooq/JooqTransacter.kt

```diff
@@ -9,14 +9,36 @@ import com.github.michaelbull.retry.policy.limitAttempts
 import com.github.michaelbull.retry.policy.plus
 import com.github.michaelbull.retry.retry
 import kotlinx.coroutines.runBlocking
+import misk.jdbc.DataSourceConfig
+import misk.jdbc.DataSourceService
+import misk.jdbc.DataSourceType
+import misk.jooq.listeners.AvoidUsingSelectStarListener
+import misk.jooq.listeners.JooqSQLLogger
+import misk.jooq.listeners.JooqTimestampRecordListener
+import misk.jooq.listeners.JooqTimestampRecordListenerOptions
+import org.jooq.Configuration
 import org.jooq.DSLContext
+import org.jooq.SQLDialect
+import org.jooq.conf.MappedSchema
+import org.jooq.conf.RenderMapping
+import org.jooq.conf.Settings
 import org.jooq.exception.DataAccessException
 import org.jooq.impl.DSL
+import org.jooq.impl.DataSourceConnectionProvider
+import org.jooq.impl.DefaultExecuteListenerProvider
+import org.jooq.impl.DefaultTransactionProvider
 import wisp.logging.getLogger
+import java.time.Clock
 import kotlin.coroutines.coroutineContext
 
 class JooqTransacter(
-  private val dslContext: Lazy<DSLContext>
+  private val dataSourceService: DataSourceService,
```

Comment:
Is this change backwards compatible? Is JooqTransacter used downstream anywhere? Looks like those call sites would break with this change.

Reply:
The JooqTransacter should be Injected in. I don't imagine anyone is instantiating a JooqTranasacter. In the off chance that someone is, this breaking change could / should force them to reconsider how they are using the lib.